internal/kafka/internal/config/kafka_supported_instance_types.go

```diff
@@ -92,10 +98,26 @@ func (k *KafkaInstanceSize) validate(instanceTypeId string) error {
 		return fmt.Errorf("maxDataRetentionPeriod for Kafka instance type '%s', size '%s' is invalid: %s", k.Id, instanceTypeId, err.Error())
 	}
 
+	maxMessageSize, err := k.MaxMessageSize.ToK8Quantity()
+	if err != nil {
+		return fmt.Errorf("maxMessageSize for Kafka instance type '%s', size '%s' is invalid: %s", k.Id, instanceTypeId, err.Error())
+	}
+
+	validSupportedAZModes := map[string]struct{}{
+		"single": {},
+		"multi":  {},
+	}
+	for _, supportedAZMode := range k.SupportedAZModes {
+		if _, ok := validSupportedAZModes[supportedAZMode]; !ok {
+			return fmt.Errorf("value '%s' in supportedAZModes for Kafka instance type '%s', size '%s' is invalid", supportedAZMode, k.Id, instanceTypeId)
+		}
+	}
+
 	if maxDataRetentionPeriod.IsZero() || egressThroughputQuantity.CmpInt64(1) < 0 ||
 		ingressThroughputQuantity.CmpInt64(1) < 0 || maxDataRetentionSize.CmpInt64(1) < 0 ||
 		k.TotalMaxConnections <= 0 || k.MaxPartitions <= 0 || k.MaxConnectionAttemptsPerSec <= 0 ||
-		k.QuotaConsumed < 1 || k.CapacityConsumed < 1 {
+		k.QuotaConsumed < 1 || k.CapacityConsumed < 1 || k.MinInSyncReplicas < 1 ||
+		k.ReplicationFactor < 1 || maxMessageSize.CmpInt64(0) < 0 || len(k.SupportedAZModes) == 0 {
```

Comment:
I think maxMessageSize should be larger than 0. wdyt of changing this check to the following?

```suggestion
		k.ReplicationFactor < 1 || maxMessageSize.CmpInt64(1) < 0 || len(k.SupportedAZModes) == 0 {
```

Reply:
I set it at 0 due to theoretically the field accepts 0 as a value: https://kafka.apache.org/documentation/#brokerconfigs_message.max.bytes

Reply:
ah didn't see that, thanks for confirming!